presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveMetadata.java

```diff
@@ -85,4 +101,52 @@ public void testCreateMixedPredicate()
 
         createPredicate(ImmutableList.of(TEST_COLUMN_HANDLE), partitions.build());
     }
+
+    @Test
+    public void testGetTablePrincipalPrivileges()
```

Comment:
What we want to test here is the conversion from `Set<HivePrivilegeInfo>` in `getTablePrincipalPrivileges()`. This is difficult because it mixes the conversion with fetching partitions from the metastore.

Instead, we can move the conversion elsewhere, such as helper method on `PrincipalPrivileges`:
```java
public static PrincipalPrivileges fromHivePrivilegeInfo(Set<HivePrivilegeInfo> privileges)
```
Then the method is easy to test without needing to mock the metastore.

Reply:
I think you can use the `FileHiveMetastore` with the `TestingHdfsEnvironment ` for this test.